src/CoreWCF.Http/src/CoreWCF/Channels/HttpTransportBindingElement.cs

```diff
@@ -325,6 +339,168 @@ public ExtendedProtectionPolicy ExtendedProtectionPolicy
-        //  public override Type MiddlewareType => typeof(ServiceModelHttpMiddleware);
+        void IPolicyExportExtension.ExportPolicy(MetadataExporter exporter, PolicyConversionContext context)
+        {
+            if (exporter == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(exporter));
+            }
+
+            if (context == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(context));
+            }
+
+            OnExportPolicy(exporter, context);
+
+            bool createdNew;
+            MessageEncodingBindingElement encodingBindingElement = FindMessageEncodingBindingElement(context.BindingElements, out createdNew);
+            if (createdNew && encodingBindingElement is IPolicyExportExtension)
+            {
+                ((IPolicyExportExtension)encodingBindingElement).ExportPolicy(exporter, context);
+            }
+
+            WsdlExporter.AddWSAddressingAssertion(exporter, context, encodingBindingElement.MessageVersion.Addressing);
+        }
+
+        internal virtual void OnExportPolicy(MetadataExporter exporter, PolicyConversionContext policyContext)
+        {
+            List<string> assertionNames = new List<string>();
+            AuthenticationSchemes effectiveAuthenticationSchemes = HttpTransportBindingElement.GetEffectiveAuthenticationSchemes(AuthenticationScheme,
+                    policyContext.BindingParameters);
+
+            if (effectiveAuthenticationSchemes != AuthenticationSchemes.None && !(effectiveAuthenticationSchemes.IsSet(AuthenticationSchemes.Anonymous)))
+            {
+                // ATTENTION: The order of the if-statements below is essential! When importing WSDL svcutil is actually
+                // using the first assertion - and the HTTP spec requires clients to use the most secure authentication
+                // scheme supported by the client. (especially important for downlevel (3.5/4.0) clients
+                if (effectiveAuthenticationSchemes.IsSet(AuthenticationSchemes.Negotiate))
+                {
+                    assertionNames.Add(TransportPolicyConstants.NegotiateHttpAuthenticationName);
+                }
+
+                if (effectiveAuthenticationSchemes.IsSet(AuthenticationSchemes.Ntlm))
+                {
+                    assertionNames.Add(TransportPolicyConstants.NtlmHttpAuthenticationName);
+                }
+
+                if (effectiveAuthenticationSchemes.IsSet(AuthenticationSchemes.Digest))
+                {
+                    assertionNames.Add(TransportPolicyConstants.DigestHttpAuthenticationName);
+                }
+
+                if (effectiveAuthenticationSchemes.IsSet(AuthenticationSchemes.Basic))
+                {
+                    assertionNames.Add(TransportPolicyConstants.BasicHttpAuthenticationName);
+                }
+
+                if (assertionNames != null && assertionNames.Count > 0)
```

Comment:
I guess this might be an extra null cautious style, but `assertionNames` is never null here